### PR TITLE
Spot robot wrapper

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR adds the first draft of a robot wrapper configuration for the BD Spot, including basic parameter definitions for legged mobile manipulators.

**TODO: camera sensors are not yet configured for Spot.**

## How Has This Been Tested
Interactive testing (**likely to be removed for final merge**):
With BD Spot robot asset downloaded to `data/robots/`, run:
 `python examples/robot_test_viewer.py`
Use `'0'` to initialize the robot and `'1-7'` to run various test functions. See the code for details. 

Unit test added (requires asset download, so won't be run on CI):

https://user-images.githubusercontent.com/1445143/141221566-9d94e967-9cff-4c13-a290-b77a7f569add.mp4

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: habitat_sim/robots/mobile_manipulator.py [hunk 1]
```diff
@@ -91,13 +99,21 @@ class MobileManipulatorParams:
...
+    leg_joints: Optional[List[int]] = None
+    leg_init_params: Optional[List[float]] = None
+
+    wheel_mtr_pos_gain: Optional[float] = None
+    wheel_mtr_vel_gain: Optional[float] = None
+    wheel_mtr_max_impulse: Optional[float] = None
+
+    leg_mtr_pos_gain: Optional[float] = None
+    leg_mtr_vel_gain: Optional[float] = None
+    leg_mtr_max_impulse: Optional[float] = None
```
Comment (ASzot) [inline habitat_sim/robots/mobile_manipulator.py:115]: I wonder if we should split the MobileManipulatorParams class apart to put the wheel stuff in the Fetch robot wrapper and the leg stuff in the Spot robot wrapper. I worry that this params class will keep growing in size as we add new robots.

File: habitat_sim/robots/mobile_manipulator.py [hunk 2]
```diff
@@ -378,13 +414,35 @@ def arm_joint_pos(self, ctrl: List[float]):
             joint_positions[self.joint_pos_indices[jidx]] = ctrl[i]
         self.sim_obj.joint_positions = joint_positions
 
-    def _validate_arm_ctrl_input(self, ctrl: List[float]):
+    @property
+    def leg_joint_pos(self) -> np.ndarray:
+        """Get the current arm joint positions."""
+        leg_pos_indices = list(
+            map(lambda x: self.joint_pos_indices[x], self.params.leg_joints)
```
Comment (Skylion007) [inline habitat_sim/robots/mobile_manipulator.py:421]: No need for map and list. This can just be a generator, it's only iterated through once.

**Review by Skylion007 — COMMENTED** [review]
Nits

**Comment by naokiyokoyama** [comment]
@aclegg3 The third joint of the Spot Arm here does not actually exist on the real robot. On the real robot, the first two joints of the arm comprise a "universal joint" at the base of the arm (see attached image), while the third one bends the elbow. In sim, the first three joints of the arm comprise a spherical joint at the base of the arm, and the fourth one bends the elbow; the ordering of the subsequent arm joints do correctly correspond with the ordering on the real robot via the BD Spot API. 

Video attached to demonstrate this sim joint that does not exist on the real robot.

Screen Shot 2021-12-05 at 4 34 56 PM
https://user-images.githubusercontent.com/5891456/144764852-00a32fb3-43cb-4209-93b4-3aa22ebfbdb6.mp4

**Comment by aclegg3** [comment]
Robot API moved to Habitat-lab.

**Pull request closed without merging**